app/src/test/java/com/x8bit/bitwarden/data/platform/processor/BridgeServiceProcessorTest.kt

```diff
@@ -1,52 +1,212 @@
+        val binder = getDefaultBinder()
+        every {
+            bridgeRepository.authenticatorSyncSymmetricKey
+        } returns SYMMETRIC_KEY.symmetricEncryptionKey.byteArray
+        assert(binder.checkSymmetricEncryptionKeyFingerprint(SYMMETRIC_KEY_FINGERPRINT))
+    }
+
+    @Test
+    @Suppress("MaxLineLength")
+    fun `getSymmetricEncryptionKeyData should return null when there is no symmetric key stored on disk`() {
+        val binder = getDefaultBinder()
+        every { bridgeRepository.authenticatorSyncSymmetricKey } returns null
+        assertNull(binder.symmetricEncryptionKeyData)
+    }
+
+    @Test
+    @Suppress("MaxLineLength")
+    fun `getSymmetricEncryptionKeyData should return the symmetric key stored on disk`() {
+        val binder = getDefaultBinder()
+        every {
+            bridgeRepository.authenticatorSyncSymmetricKey
+        } returns SYMMETRIC_KEY.symmetricEncryptionKey.byteArray
+        assertEquals(SYMMETRIC_KEY, binder.symmetricEncryptionKeyData)
+    }
+
+    @Nested
+    inner class SyncAccountsTest {
+
+        private var lastAccountsSync: EncryptedSharedAccountData? = null
+
+        private val serviceCallback = object : IBridgeServiceCallback.Stub() {
+            override fun onAccountsSync(data: EncryptedSharedAccountData?) {
+                lastAccountsSync = data
+            }
+        }
+
+        @BeforeEach
+        fun setup() {
+            // Setup RemoteCallbackList to call back to serviceCallback:
+            mockkConstructor(RemoteCallbackList::class)
+            every {
+                anyConstructed<RemoteCallbackList<IBridgeServiceCallback>>()
+                    .register(serviceCallback)
+            } returns true
+            every {
+                anyConstructed<RemoteCallbackList<IBridgeServiceCallback>>()
+                    .beginBroadcast()
+            } returns 1
+            every {
+                anyConstructed<RemoteCallbackList<IBridgeServiceCallback>>()
+                    .getBroadcastItem(0)
+            } returns serviceCallback
+            lastAccountsSync = null
+        }
+
+        @Test
+        fun `syncAccounts when symmetricEncryptionKeyData is null should do nothing`() {
+            every { bridgeRepository.authenticatorSyncSymmetricKey } returns null
+            getDefaultBinder().syncAccounts()
+            assertNull(lastAccountsSync)
+        }
+
+        @Test
+        fun `syncAccounts should encrypt result from BridgeRepository`() {
+            val sharedAccountData = mockk<SharedAccountData>()
+            val expected = mockk<EncryptedSharedAccountData>()
+            every {
+                bridgeRepository.authenticatorSyncSymmetricKey
+            } returns SYMMETRIC_KEY.symmetricEncryptionKey.byteArray
+            coEvery { bridgeRepository.getSharedAccounts() } returns sharedAccountData
+            mockkStatic(SharedAccountData::encrypt)
```

Comment:
This should also be unmocked in the tearDown

Reply:
Got u. e9863c266fb2e2f6e132bad665ae34182c8e9649